### PR TITLE
Introduced request types

### PR DESCRIPTION
**Pull request by jeongkyun-oh**
## Proposed changes

- `requestType` is introduced.
- This PR should apply the changes after https://github.com/klaytn/klaytn/pull/602 merged.

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [x] New feature or enhancement
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [x] Lint and unit tests pass locally with my changes (`$ make test`)
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: datasync/chaindatafetcher/types.go [hunk 1]
```diff
@@ -34,9 +43,40 @@ type request struct {
...
+	return rt&requestTypeTokenTransfer == requestTypeTokenTransfer
+}
+
+func hasContracts(rt requestType) bool {
+	return rt&requestTypeContracts == requestTypeContracts
+}
+
+func hasTraces(rt requestType) bool {
+	return rt&requestTypeTraces == requestTypeTraces
+}
```
Comment (aidan-kwon) [inline datasync/chaindatafetcher/types.go:60]: These functions can be expressed as a function.
```
func checkReqeustType(rt reqeustType, tartgetType requestType) bool {
    return rt&targetType == targetType 
}
```

File: datasync/chaindatafetcher/types_test.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+		{true, true, true, false},
+		{true, true, false, true},
+		{false, true, true, true},
+
+		// all type
+		{true, true, true, true},
+	}
+
+	checkFuncs := []func(requestType) bool{
+		hasTransactions, hasTokenTransfers, hasTraces, hasContracts,
```
Comment (aidan-kwon) [inline datasync/chaindatafetcher/types_test.go:74]: Please, change this line.

**Comment by jeongkyun-oh** [comment]
@KimKyungup PTAL

File: datasync/chaindatafetcher/chaindata_fetcher.go [hunk 3]
```diff
@@ -245,15 +242,30 @@ func (f *ChainDataFetcher) handleRequest() {
...
+				if err := f.repo.InsertContracts(req.event); err != nil {
+					f.resCh <- newResponse(req.reqType, req.event.Block.Number(), err)
+				}
+			}
+
+			if checkRequestType(req.reqType, requestTypeTraces) {
+				if err := f.repo.InsertTraceResults(req.event); err != nil {
+					f.resCh <- newResponse(req.reqType, req.event.Block.Number(), err)
+				}
+			}
```
Comment (KimKyungup) [inline datasync/chaindatafetcher/chaindata_fetcher.go:268]: How about send one newResponse instead of several newResponses?

**Pull request merged**